drv/sidecar-mainboard-controller-api/src/tofino2.rs

```diff
@@ -0,0 +1,143 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::{Addr, Reg};
+use drv_fpga_api::{FpgaError, FpgaUserDesign, WriteOp};
+use userlib::FromPrimitive;
+use zerocopy::AsBytes;
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum TofinoSeqState {
+    Initial = 0,
+    A2 = 1,
+    A0 = 2,
+    InPowerUp = 3,
+    InPowerDown = 4,
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum TofinoSeqError {
+    None = 0,
```

Comment:
You can remove `None` and use `Option<TofinoSeqError>` below; Rust is smart enough to still pack it into a single byte ([test](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ccef8f80f580dd619f81bb0f80261cf7))

Reply:
The sequencer server wants to pass `TofinoSeqError` though an Idol interface. The `None` variant lets me do this without much fuss. Can `Option<TofinoSeqError>` likewise pass through Idol without much additional work?